Lib/fontTools/feaLib/lexer.py

```diff
@@ -45,13 +47,13 @@ def __init__(self, text, filename):
     def __iter__(self):
         return self
 
-    def next(self):  # Python 2
-        return self.__next__()
+    def next(self, comments=False):  # Python 2
```

Comment:
I have the following text in my rendering of that page:


iterator.__next__()

    Return the next item from the container. If there are no further items, raise the StopIteration exception. This method corresponds to the tp_iternext slot of the type structure for Python objects in the Python/C API.

No mention of extra parameters, which would obviously have to be named parameters. It also doesn't mention that in normal cases you never call the __next__ method directly.

I really don't want to go down the road of constantly switching the state of the Lexer. Hmm, in the case of the parser, if I did that, then I would turn on passing comments, and perhaps not have the option to turn it off. Is that preferred?